src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1110,231 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
+
+      // named buffers always have offset 0
+      SLIC_ASSERT_MSG(bv->getSchema().dtype().offset() == 0, "");
+      SLIC_ASSERT_MSG(bv->getNumElements() >= offset + numDofs, "");
+
+      if (vv->isEmpty())
+      {
+         vv->attachBuffer(bv->getBuffer())
+         ->apply(sidre::DOUBLE_ID, numDofs, offset);
+      }
+
+      qf->NewDataAndSize(vv->getData(), numDofs);
+   }
+   else
+   {
+      // If we are not managing the grid function's data,
+      // create a view with the external data
+      vv->setExternalDataPtr(sidre::DOUBLE_ID, numDofs, qf->GetData());
+   }
+   SLIC_ASSERT_MSG((numDofs > 0 && vv->isApplied()) ||
+                   (numDofs == 0 && vv->isEmpty() && vv->isDescribed()),
+                   "invalid View state");
+   SLIC_ASSERT_MSG(numDofs == 0 || vv->getData() == qf->GetData(),
+                   "View data is different from QuadratureFunction data");
+   SLIC_ASSERT_MSG(vv->getNumElements() == numDofs,
+                   "View size is different from QuadratureFunction size");
+}
+
+// private method
+// Should only be called on mpi rank 0 ( or if serial problem ).
+void MFEMSidreDataCollection::
+RegisterQFieldInBPIndex(const std::string& field_name, mfem::QuadratureFunction *qf)
+{
+   sidre::Group *bp_field_grp = m_bp_grp->getGroup("qfields/" + field_name);
+   sidre::Group *bp_index_field_grp =
+      m_bp_index_grp->createGroup("qfields/" + field_name);
+
+   bp_index_field_grp->createViewString( "path", bp_field_grp->getPathName() );
+   bp_index_field_grp->copyView( bp_field_grp->getView("topology") );
+   bp_index_field_grp->copyView( bp_field_grp->getView("vdim") );
+   bp_index_field_grp->copyView( bp_field_grp->getView("qspace_order") );
+
+   // Fix me?: Should this be qf->GetVDim() instead?
+   // It's not clear to me what this exactly does, since the value 1 works just fine
+   // even when qf->GetVDim() > 1.
+   const int number_of_components = 1;
+   bp_index_field_grp->createViewScalar("number_of_components",
+                                        number_of_components);
+}
+
+void MFEMSidreDataCollection::RegisterQField(const std::string &field_name,
+                                        mfem::QuadratureFunction *qf,
+                                        const std::string &buffer_name,
+                                        axom::sidre::IndexType offset)
+{
+   if ( field_name.empty() || buffer_name.empty() ||
+        qf == NULL || qf->GetSpace() == NULL )
+   {
+#ifdef AXOM_DEBUG
+      SLIC_WARNING("QField with the name '" << field_name << "' was provided a null"
+                  " QuadratureFunction, so nothing was done.");
+#endif
+      return;
+   }
+
+   // Register field_name in the blueprint group.
+   sidre::Group* f = m_bp_grp->getGroup("qfields");
+
+   if (f->hasGroup( field_name ))
+   {
+      // There are two possibilities:
+      // 1. If HasField(field_name) is true - we are overwriting a field that
```

Comment:
typo in comment: `HasField` -> `HasQField`

Reply:
Fixed